base/array.jl

```diff
@@ -711,8 +711,8 @@ function find(testf::Function, A::AbstractArray)
     # use a dynamic-length array to store the indexes, then copy to a non-padded
     # array for the return
     tmpI = Array(Int, 0)
-    for i = 1:length(A)
-        if testf(A[i])
+    for (i,j) = enumerate(eachindex(A))
+        if testf(A[j])
```

Comment:
```jl
julia> A = rand(5)
5-element Array{Float64,1}:
 0.819467
 0.980514
 0.190881
 0.936337
 0.882508

julia> for (i, j) in enumerate(A)
           @show A[j]
       end
WARNING: Indexing with non-Integer Reals is deprecated.  It may be that your index arose from an integer division of the form i/j, in which case you should consider using i÷j or div(i,j) instead.
 in depwarn at deprecated.jl:73
 in to_index at deprecated.jl:447
 in getindex at array.jl:282
 [inlined code] from show.jl:127
 in anonymous at no file:0
while loading no file, in expression starting on line 0
ERROR: InexactError()
 in to_index at deprecated.jl:448
 in getindex at array.jl:282
 [inlined code] from show.jl:127
 in anonymous at no file:0

julia> for (i, a) in enumerate(A)
           @show a
       end
a = 0.8194665233152139
a = 0.9805141257931855
a = 0.19088128894552314
a = 0.9363374114892509
a = 0.8825084643072016
```